src/utils/AppRouteProvider.jsx

```diff
@@ -10,9 +15,15 @@ const AppRouterContext = React.createContext(null);
 const AppRouteProvider = (props) => {
   const { children } = props;
   const router = useRouter();
+  const dispatch = useDispatch();
 
   const [renderIntercept, setRenderIntercept] = useState(null);
 
+  const activeProjectUuid = useSelector((state) => state?.projects?.meta?.activeProjectUuid);
```

Comment:
if this is only going to be used when `previousRoute.match('/data-management')` then it should be put inside the if. Here this will be run for every request.

Reply:
We can't do this because hooks can only be called from the top level of a functional component, not inside a loop or condition (https://reactjs.org/docs/hooks-rules.html). Trying to move this into the function definition causes an error.

Reply:
Right now, changes to `experiments`, `activeProjectUuid` or `experimentId` causes this element to re-render. This would happen only on the data-management page. In the other modules, users do not change experiments or projects, so this would be fine.

I checked the profiler, and yes change to these variables cause re-rendering of the app provider. However, changing these variables will not only impact this component, but also other components wrapped by the `provider` which displays the project and experiments visually. I checked in the `Profiler` and the impact of re-rendering of this component is small compared to the other components.

![image](https://user-images.githubusercontent.com/2862362/149803632-26eb8046-e507-40f5-b99e-07bb590ed60a.png)